File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/search/model/company/SearchCompany.java
package dev.reuss.tmdb.domain.search.model.company;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Company returned by TMDB company search.
 *
 * @param id            TMDB company id
 * @param logoPath      company logo image path
 * @param name          company name
 * @param originCountry origin country code
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record SearchCompany(
        int id,

        @JsonProperty("logo_path")
        String logoPath,

        String name,

        @JsonProperty("origin_country")
        String originCountry
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/search/model/company/SearchCompany.kt
package dev.reuss.tmdb.domain.search.model.company

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Company returned by TMDB company search.
 *
 * @property id TMDB company id
 * @property logoPath TMDB image path for the logo.
 * @property name company name
 * @property originCountry ISO 3166-1 code for the resource's country of origin.
 */
@JvmRecord
data class SearchCompany(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("logo_path")
    val logoPath: String?,
    @all:JsonProperty("name")
    val name: String?,
    @all:JsonProperty("origin_country")
    val originCountry: String?,
) : TmdbModel
